ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/context/KsqlRestServiceContextBinder.java

```diff
@@ -31,9 +35,19 @@
 public class KsqlRestServiceContextBinder extends AbstractBinder {
   public KsqlRestServiceContextBinder(
       final KsqlConfig ksqlConfig,
-      final KsqlSecurityExtension securityExtension
+      final KsqlSecurityExtension securityExtension,
+      final Supplier<SchemaRegistryClient> schemaRegistryClientSupplier
   ) {
-    KsqlRestServiceContextFactory.configure(ksqlConfig, securityExtension);
+    KsqlRestServiceContextFactory.configure(ksqlConfig, securityExtension,
+        schemaRegistryClientSupplier);
+  }
+
+  public static BiFunction<KsqlConfig, KsqlSecurityExtension, Binder> createFactory(
```

Comment:
If we're adding a factory method, can we make the constructor private or package private?

Reply:
Done